apps/tasks/src/jobs/integrations/media-organizer.ts

```diff
@@ -0,0 +1,55 @@
+import dayjs from "dayjs";
+import SuperJSON from "superjson";
+
+import { decryptSecret } from "@homarr/common";
+import { db, eq } from "@homarr/db";
+import { items } from "@homarr/db/schema/sqlite";
+import { SonarrIntegration } from "@homarr/integrations";
+import { createCacheChannel } from "@homarr/redis";
+import type { WidgetComponentProps } from "@homarr/widgets";
+
+import { EVERY_MINUTE } from "~/lib/cron-job/constants";
+import { createCronJob } from "~/lib/cron-job/creator";
+
+export const mediaOrganizerJob = createCronJob(EVERY_MINUTE).withCallback(async () => {
+  const itemsForIntegration = await db.query.items.findMany({
+    where: eq(items.kind, "calendar"),
+    with: {
+      integrations: {
+        with: {
+          integration: {
+            with: {
+              secrets: {
+                columns: {
+                  kind: true,
+                  value: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  for (const itemForIntegration of itemsForIntegration) {
+    for (const integration of itemForIntegration.integrations) {
```

Comment:
As already mentioned this will fetch for all integrations multiple times if items with the same integration exist, is this intentional?

Reply:
No. Will fix 👍

Reply:
How do we deal with it if there are multiple widgets (items) that use this integration but with different options?
I would propose the following code for this Cron Job:
![image](https://github.com/homarr-labs/homarr/assets/30572287/9c980aef-db99-4211-bc5b-e518deaf23f7)
Simple options like numbers could be reduced to the max and then we could calculate in each procedure and exclude depending the item options - but that's quite a lot of effort.
What do you think?

Reply:
Maybe we should sent one per widget and one per integration instead? Distinct sadly also doesn't solve the problem (messages will be overwritten)

Reply:
Oh yeah you're right. Probably best to do so